src/actions/auger/auger_train.ts

```diff
@@ -0,0 +1,490 @@
+          workers_count: 2,
+          cluster_autoterminate_minutes: 10,
+          token: transaction.token,
+        },
+        json: true,
+        resolveWithFullResponse: true,
+      }
+      return httpRequest(options).promise()
+    } catch (e) {
+      throw new Error(`project not started ${e}`)
+    }
+  }
+
+  private async startProjectFile(transaction: Transaction): Promise<any> {
+    const projectFile = await this.createProjectFile(transaction)
+    transaction.successStatus = "processed"
+    transaction.errorStatus = "processed_with_error"
+    transaction.pollFunction = this.getProjectFile.bind(this)
+    if (transaction.params.train_data === "true") {
+      transaction.callbackFunction = this.startExperiment.bind(this)
+    } else {
+      transaction.callbackFunction = undefined
+    }
+    transaction.projectFileId = projectFile.body.data.id
+    this.startPoller(transaction)
+  }
+
+  private async startExperiment(transaction: Transaction): Promise<any> {
+    const experiment = await this.createExperiment(transaction)
+    transaction.experimentId = experiment.body.data.id
+    await this.startSession(transaction)
+  }
+
+  private async createProjectFile(transaction: Transaction): Promise<any> {
+    try {
+      const projOptions = {
+        url: `${transaction.augerURL}/projects/${transaction.projectId}/files/`,
+        json: true,
+        resolveWithFullResponse: true,
+        body: {
+          project_id: transaction.projectId,
+          file_name: `${transaction.fileName}.json`,
+          name: transaction.fileName,
+          url: transaction.s3Path,
+          content_type: transaction.contentType,
+          token: transaction.token,
+        },
+      }
+      return httpRequest.post(projOptions).promise()
+    } catch (e) {
+      throw new Error(`project file not created: ${e}`)
+    }
+  }
+
+  private async createExperiment(transaction: Transaction): Promise<any> {
+    try {
+      const options = {
+        url: `${transaction.augerURL}/experiments/`,
+        body: {
+          project_id: transaction.projectId,
+          name: `${transaction.fileName}_${Date.now()}`,
+          project_file_id: transaction.projectFileId,
+          token: transaction.token,
+        },
+        json: true,
+        resolveWithFullResponse: true,
+      }
+      return httpRequest.post(options).promise()
+    } catch (e) {
+      throw new Error(`experiment start failed: ${e}`)
+    }
+
+  }
+
+  private async getProjectFile(transaction: Transaction): Promise<any> {
+    try {
+      const options = {
+        url: `${transaction.augerURL}/project_files/${transaction.projectFileId}?token=${transaction.token}`,
+        json: true,
+        resolveWithFullResponse: true,
+      }
+      return httpRequest.get(options).promise()
+    } catch (e) {
+      throw new Error(`project file fetched: ${e}`)
+    }
+  }
+
+  // private async getExperiment(transaction: Transaction): Promise<any> {
```

Comment:
Is this commented function needed anymore?

Reply:
I'll remove it